lua/telescope/config/resolve.lua

```diff
@@ -88,46 +93,81 @@ That's the next step to scrolling.
+local _resolve_map = {}
 
-local _resolve_map = {
-  -- Booleans
-  [function(val) return val == false end] = function(selector, val)
-    return function(...)
-      return val
-    end
-  end,
+-- Booleans
+_resolve_map[function(val) return val == false end] = function(_, val)
+  return function(...)
+    return val
+  end
+end
 
-  -- Percentages
-  [function(val) return type(val) == 'number' and val >= 0 and val < 1 end] = function(selector, val)
-    return function(...)
-      local selected = select(selector, ...)
-      return math.floor(val * selected)
-    end
-  end,
+-- Percentages
+_resolve_map[function(val) return type(val) == 'number' and val >= 0 and val < 1 end] = function(selector, val)
+  return function(...)
+    local selected = select(selector, ...)
+    return math.floor(val * selected)
+  end
+end
 
-  -- Numbers
-  [function(val) return type(val) == 'number' and val >= 1 end] = function(selector, val)
-    return function(...)
-      local selected = select(selector, ...)
-      return math.min(val, selected)
-    end
-  end,
+-- Numbers
+_resolve_map[function(val) return type(val) == 'number' and val >= 1 end] = function(selector, val)
+  return function(...)
+    local selected = select(selector, ...)
+    return math.min(val, selected)
+  end
+end
 
+-- Tables TODO:
+-- ... {70, max}
 
-  -- Tables TODO:
-  -- ... {70, max}
+-- function:
+--    Function must have same signature as get_window_layout
+--        function(self, max_columns, max_lines): number
+--
+--    Resulting number is used for this configuration value.
+_resolve_map[function(val) return type(val) == 'function' end] = function(_, val)
+  return val
+end
 
+-- Add padding option
+_resolve_map[function(val) return type(val) == 'table' and val['padding'] ~= nil end] = function(selector, val)
+  local resolve_pad = function(value)
+    for k, v in pairs(_resolve_map) do
+      if k(value) then
+        return v(selector, value)
+      end
+    end
+
+    error('invalid configuration option for padding:' .. tostring(value))
+  end
+
+  return function(...)
+    local selected = select(selector, ...)
+    local padding = resolve_pad(val['padding'])
+    return math.floor(selected - 2 * padding(...))
+  end
+end
 
-  -- function:
-  --    Function must have same signature as get_window_layout
-  --        function(self, max_columns, max_lines): number
-  --
-  --    Resulting number is used for this configuration value.
-  [function(val) return type(val) == 'function' end] = function(selector, val)
-    return val
-  end,
-}
 
+--- Converts input to a function that returns the height.
+--- The input must take one of four forms:
+--- 1. 0 <= number < 1 <br>
+---     This means total height as a percentage.
+--- 2. 1 <= number <br>
+---     This means total height as a fixed number.
+--- 3. function <br>
+---     Must have signature:
+---       function(self, max_columns, max_lines): number
+--- 4. table of the form:
+---       {padding = `foo`} <br>
```

Comment:
This resolves in "stupid" `<br>` in the docgen output. I never said my formatter is prefect :rofl: but these two lines will wrap anyways and the issue is fixed if you put the `{padding = `foo`} <br>` immediatly after the `:`